apis/stacks/v1alpha1/types.go

```diff
@@ -109,6 +119,21 @@ func (si *StackInstall) PermissionScope() string { return string(apiextensions.N
 // PermissionScope gets the required app.yaml permissionScope value ("Cluster") for ClusterStackInstall
 func (si *ClusterStackInstall) PermissionScope() string { return string(apiextensions.ClusterScoped) }
 
+// ControllerNamespace is the namespace where a Stack's StackInstall Job,
+// Controller, ServiceAccount, and namespaced RBAC roles will be created.
+func (si *StackInstall) ControllerNamespace() string { return si.Namespace }
+
+// ControllerNamespace is the namespace where a Stack's ClusterStackInstall Job,
+// Controller, and ServiceAccount will be created.
+func (si *ClusterStackInstall) ControllerNamespace() string {
+	if si.Spec.ControllerNamespace == "" {
+		// TODO(displague) make this a stack-manager CLI and ENV param or return
+		// namespace of the stack-manager (preventing namespace DNE problems)
+		return "crossplane-system"
```

Comment:
Could we not just make `.spec.controllerNamespace` a required field and avoid setting a default?

Reply:
I was trying to keep the "ClusterStackInstall" `spec` simple and compatible with existing documentation.

Making the variable explicit now doesn't prevent us from defaulting it later.

Are there other compelling reasons to not use a default, or specifically "crossplane-system"?
Would the implementation of the `TODO` comment help?

Reply:
I typically prefer to avoid defaulting in favor of convention. This is what we do for Crossplane Services types; I don't believe `crossplane-system` is hardcoded anywhere in Crossplane's codebase but we use it by convention in all examples and documentation for any type that can't function without a namespace specified (`writeConnectionSecretToRef` etc).

The implementation of the TODO would be better than hard coding this default value, but I feel it would still be a worse user experience than making the field mandatory. When the field is mandatory the mental model for setting and getting the value is always the same: read or write the `controllerNamespace` of your manifest. When it's optional with a default (set somewhere else within Crossplane) the mental model becomes more complicated. It's what you set in your YAML, unless it's not, in which case you (as a user who may not be able to see the stack manager pod) need to go somehow figure out what env var or flag Crossplane is running with. Or worse, just submit your YAML and see what happens.

Reply:
What happens to existing ClusterStackInstalls upon updating Crossplane to a version that now requires this field?  Will users even be permitted to update the CRD in this case if existing CRs are not considered valid against it?

Is there an opportunity for the StackManager to 'heal' existing ClusterStackInstalls by setting an empty controllerNamespace using the existing controller reference?

Are these non-concerns for our current stability level?